In [ ]:
import pandas as pd
import numpy as np
import scipy.stats as stats

## H0 --> There is no significant association between the type of device purchased and the customer satisfaction level
## Ha --> There is significant association between the type of device purchased and the customer satisfaction level

In [ ]:
Thermostat=[50,80,60,30,20]
Smartlight=[70,100,90,50,50]
alpha=0.05
dof=5-1
dof

4

In [ ]:
contingency_table = np.array([Thermostat, Smartlight])
contingency_table

array([[ 50,  80,  60,  30,  20],
       [ 70, 100,  90,  50,  50]])

In [ ]:
chi2_stat,p_value, dof, expected_val = stats.chi2_contingency(contingency_table)
print('chi_square statistics:',chi2_stat)
print('p_value:',p_value)
print('degree of freedom:',dof)
print('expected values:',expected_val)

chi_square statistics: 5.638227513227513
p_value: 0.22784371130697179
degree of freedom: 4
expected values: [[ 48.  72.  60.  32.  28.]
 [ 72. 108.  90.  48.  42.]]


In [ ]:
chi_critical = stats.chi2.ppf(0.95,4)
chi_critical

np.float64(9.487729036781154)

## chi_critical > chi_stats then failed to reject H0
## 9.48 is greater than 5.638 so it is failed to reject H0, it means there is no assoication between the type of device purchased and the customer satisfaction level
## the Device pUrchased and Customer Satisfaction levels are independent.


In [ ]:
## z- Distribution table

## H0 --> cost of the model is accurate, weekly operating costs are higher.
## Ha --> cost of the model is not accurate

In [ ]:
n=25
x_bar=3050
pop_mean=1000+5*600
std_dev=5*25
alpha=0.05

z_score=(x_bar-pop_mean)/(std_dev/np.sqrt(n))
z_score

np.float64(-38.0)

In [ ]:
z_critcal=stats.norm.ppf(1-(alpha/2))
z_critcal

np.float64(1.959963984540054)

In [ ]:
## z_scrore is a negative number then z_critical is also considered as negative number, that is -1.959

## z_critical < z_score then reject H0
## -1.95 > -38.0 so failed to reject H0.
## It means cost of the model is accurate --> weekly operating costs are higher..